source/core/GenericFunctions.cpp

```diff
@@ -21,12 +21,20 @@ namespace Vireo
 {
 
 //------------------------------------------------------------
-ConstCStr CopyProcName(void *pSource, void *pDest, Int32 aqSize)
+ConstCStr CopyProcName(void *pSource, void *pDest, Int32 aqSize, Boolean isEnum)
 {
     // If the source or dest are not aligned to aqSize bytes, return null.
     if (((uintptr_t) pSource % aqSize != 0) || ((uintptr_t) pDest % aqSize != 0))
         return null;
-
+    if (isEnum) {
+        switch (aqSize) {
+            case 1:     return "CopyEnum1";   break;
+            case 2:     return "CopyEnum2";   break;
+            case 4:     return "CopyEnum4";   break;
+            case 8:     return "CopyENum8";   break;
```

Comment:
I'm curious why we don't have CopyEnum16 and CopyEnum32. I think a comment explaining why they're not necessary is helpful here, otherwise the code feels that is incomplete.